draft-xie-bidirectional-messaging-latest.md

```diff
@@ -30,63 +30,74 @@ author:
+use in this type of application. This draft proposes an HTTP/2 protocol
+extension that enables bidirectional communication between client and server.
+
+Currently, the only mechanism in HTTP/2 for server to client communication is
 server push. That is, servers can initiate unidirectional push promised streams
-to clients, but clients cannot respond to them, except accepting or discarding
-them silently. More interesting, the intermediaries may have different server
-push policies, and may not forward to downstream at all. The best effort
-mechanism is not reliable to deliver content from servers to clients. While this
-satisfies some use-cases, its unidirectional property limits HTTP/2 for wider
-use, for example, send messages and notifications from servers to clients at the
-time they are available.
-
-To work around this limitation, many techniques are developed, like long polling
-{{!RFC6202}}, WebSocket {{!RFC8441}}, and tunneling. They are common at:
-layering application protocol on top of HTTP/2 and using HTTP/2 streams as
-transport connections. These solutions more or less defeat the optimizations
-provided by HTTP/2. For example, first, if multiplexing multiple parallel
-interactions into one HTTP/2 stream, the head of line block will be
-re-introduced. Second, application meta data is encapsulated into DATA frame,
-rather than HEADERS frame, so header compression is impossible. Third, user data
-are framed more than once at different protocol layers, which offsets the wire
-efficiency from HTTP/2 binary frame. Take WebSocket over HTTP/2 as an example,
-user data is framed at application protocol, WebSocket, and HTTP/2 layers. This
-not only introduces the byte overhead on the wire, but also complicates the data
-processing. These techniques also pose new operational challenges to
-intermediaries. Due to the traffic from the whole user session is encapsulated
-into one HTTP/2 stream, this stream can last very long time. Intermediaries may
-take long time to drain these streams. As a remind, GOAWAY can only signals the
-remote endpoint to stop using the connection for new streams and does not help
-the open streams. Moreover, intermediaries have no visibility to user
-interactions, and lose the capability to collect telemetry metrics (e.g., time
-to the first/last byte of request and response) for services.
+to clients, but clients cannot respond to them and either accept or discard
+them silently. Additionally, intermediaries along the path may have different
+server push policies and may not forward push promised streams to the downstream
+client. This best effort mechanism is not sufficient to reliably deliver content
+from servers to clients, limiting additional use-cases, such as sending messages
+and notifications from servers to clients immediately when they become
+available.
+
+To work around this limitation, several techniques are used, like long polling
```

Comment:
This is just wording and splits it into two paragraphs, not sure why the diff shows it as all different...